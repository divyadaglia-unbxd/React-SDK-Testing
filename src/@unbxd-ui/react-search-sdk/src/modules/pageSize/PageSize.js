import React from 'react';
import PropTypes from 'prop-types';

import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';
import PageSizeContainer from './PageSizeContainer';

/**
 * Component to manage the page size.
 */
const PageSize = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(PageSize.displayName);
                }

                const {
                    unbxdCore,
                    unbxdCoreStatus,
                    unbxdState,
                    helpers
                } = appState;
                const { pageSize } = unbxdState;
                return (
                    <PageSizeContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        helpers={helpers}
                        pageSize={pageSize}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

PageSize.displayName = 'PageSize';
PageSize.defaultProps = {
    size: 10,
    sizeOptions: [
        { id: 5, value: '5' },
        { id: 10, value: '10' },
        { id: 15, value: '15' }
    ],
    displayType: 'DROPDOWN'
};

PageSize.propTypes = {
    /**
     * Number of products to load on a page.
     */
    size: PropTypes.number,
    /**
     * Options of number of products to load on a page.
     */
    sizeOptions: PropTypes.arrayOf(
        PropTypes.shape({ id: PropTypes.number, value: PropTypes.string })
    ),
    /**
     * Display type of `DROPDOWN` or `LIST` for pageSize.
     */
    displayType: PropTypes.string,
    /**
     * Custom size component instance.
     */
    pageSizeItemComponent: PropTypes.element,
    /**
     * Label for the module.
     */
    label: PropTypes.node
};

export default PageSize;
