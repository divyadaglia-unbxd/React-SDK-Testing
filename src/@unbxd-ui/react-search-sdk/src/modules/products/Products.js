import React from 'react';
import PropTypes from 'prop-types';

import ProductsContainer from './ProductsContainer';
import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';

/**
 * Component to manage products.
 * Products also manages the pagination options for the search results here.
 */
const Products = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(Products.displayName);
                }

                const {
                    unbxdCore,
                    unbxdCoreStatus,
                    helpers,
                    unbxdState,
                    priceUnit
                } = appState;
                const { viewType, pageSize, sort } = unbxdState;

                return (
                    <ProductsContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        helpers={helpers}
                        viewType={viewType}
                        pageSize={pageSize}
                        sort={sort}
                        priceUnit={priceUnit}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

Products.displayName = 'Products';

Products.defaultProps = {
    perRow: 4,
    pageSize: 10,
    variantsCount: 0,
    paginationType: 'FIXED_PAGINATION',
    heightDiffToTriggerNextPage: 300,
    showLoader: false,
    productIdAttribute: 'uniqueId',
    groupBy: '',
    variantAttributesMap: {},
    onResultsLoaded:()=>{}
};

Products.propTypes = {
    /**
     * Number of products to be shown per row.
     */
    perRow: PropTypes.number,
    /**
     * Required pagination type. Possible options are `INFINITE_SCROLL`, `CLICK_N_SCROLL` and `FIXED_PAGINATION`.
     */
    paginationType: PropTypes.string,
    /**
     * Height difference to trigger for next page in case of paginationType `INFINITE_SCROLL`.
     */
    heightDiffToTriggerNextPage: PropTypes.number,
    /**
     * Custom load more component instance
     */
    loadMoreComponent: PropTypes.element,
    /**
     * Unique attribute of the product
     */
    productIdAttribute: PropTypes.string,
    /**
     * Mapping of catalog Product fields to SDK Product fields.
     */
    attributesMap: PropTypes.object.isRequired,
    /**
     * Show variants to a product.
     */
    showVariants: PropTypes.bool,
    /**
     * Number of variants to fetch.
     */
    variantsCount: PropTypes.number,
    /**
     * Mapping of catalog Product variant fields to SDK Product variant fields.
     */
    variantAttributesMap: PropTypes.object,
    /**
     * Should loader be shown
     */
    showLoader: PropTypes.bool,
    /**
     * Custom loader component instance
     */
    loaderComponent: PropTypes.element,
    /**
     * Callback function triggered on click of a product.
     */
    onProductClick: PropTypes.func,
    /**
     * Callback function triggered on zero results.
     */
    onZeroResults: PropTypes.func,
    /**
     * Display swatches
     */
    showSwatches: PropTypes.bool,
    /**
     * Group variants by the attribute
     */
    groupBy: PropTypes.string,
    /**
     * Swatch attributes that change on click on of the swatch
     */
    swatchAttributesMap: PropTypes.object,
    /**
     * Custom swatch component instance
     */
    swatchItemComponent: PropTypes.element,
    /**
     * Custom product item component instance
     */
    productItemComponent: PropTypes.element,
    /**
     *  Custom zero result component instance
     */
    zeroResultsComponent: PropTypes.element,
    /**
     *  callback on each new results load
     */
     onResultsLoaded: PropTypes.func
};

export default Products;
