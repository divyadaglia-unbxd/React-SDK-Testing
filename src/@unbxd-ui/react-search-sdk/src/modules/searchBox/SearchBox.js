import React from 'react';
import PropTypes from 'prop-types';

import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';
import SearchBoxContainer from './SearchBoxContainer';

/**
 * Component to display merchandising banners.
 */
const SearchBox = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(SearchBox.displayName);
                }

                const {
                    unbxdCore,
                    unbxdCoreStatus,
                    helpers,
                    productType,
                    unbxdState
                } = appState;

                const { query } = unbxdState;

                return (
                    <SearchBoxContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        helpers={helpers}
                        productType={productType}
                        query={query}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

SearchBox.displayName = 'SearchBox';

SearchBox.defaultProps = {
    autoFocus: false,
    clearable: false,
    showLoader: false
};

SearchBox.propTypes = {
    /**
     * Should the searchbox be focused by default.
     */
    autoFocus: PropTypes.bool,
    /**
     * Should the searchbox be clearable.
     */
    clearable: PropTypes.bool,
    /**
     * Hook for search query. The function should return `true` if the search is to be triggered, false otherwise.
     */
    onSubmit: PropTypes.func,
    /**
     * Hook for clearing the search query. The function should return `true` if the searchbox is to be cleared, false otherwise.
     */
    onClear: PropTypes.func,
    /**
     * Should loader be shown
     */
    showLoader: PropTypes.bool,
    /**
     * Custom input component instance
     */
    inputComponent: PropTypes.element,
    /**
     * Custom submit component instance
     */
    submitComponent: PropTypes.element,
    /**
     * Custom reset component instance
     */
    clearComponent: PropTypes.element,
    /**
     * Default search query
     */
    defaultSearch: PropTypes.string,
    /**
     * SearchBox placeholder
     */
    placeholder: PropTypes.string
};

export default SearchBox;
