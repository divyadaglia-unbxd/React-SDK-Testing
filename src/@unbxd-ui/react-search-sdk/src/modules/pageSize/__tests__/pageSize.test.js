import renderer from 'react-test-renderer';
import { render, waitFor, fireEvent, screen } from '@testing-library/react';
import React from 'react';
import Products from '../../products';
import UnbxdSearchWrapper from '../../../UnbxdSearchWrapper';
import PageSize from '../index';
import { size5SearchResponse } from './mocks';
import SearchBox from '../../searchBox';
import { searchResponse } from '../../products/__tests__/mocks/searchMock';

// establish API mocking before all tests
beforeAll(() => {
    window.fetch = jest.fn((request) => {
        if (request.includes('rows=5')) {
            return Promise.resolve({
                json: () => Promise.resolve(size5SearchResponse)
            });
        }
        return Promise.resolve({
            json: () => Promise.resolve(searchResponse)
        });
    });
});

const attributesMap = {
    title: 'title',
    uniqueId: 'uniqueId',
    imageUrl: 'imageUrl',
    price: 'RRP_Price',
    productUrl: 'productUrl'
};

test('Match Snapshot for page size', async () => {
    const tree = renderer.create(
        <UnbxdSearchWrapper
            siteKey="wildearthclone-neto-com-au808941566310465"
            apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
        >
            <PageSize />
            <SearchBox defaultSearch="shoes" />
            <Products attributesMap={attributesMap} />
        </UnbxdSearchWrapper>
    );
    await waitFor(() => tree.toJSON());
    expect(tree.toJSON()).toMatchSnapshot();
});

test('Page size test', async () => {
    const { getByText, getByTestId, container } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <PageSize />
                <SearchBox defaultSearch="shoes" />
                <Products attributesMap={attributesMap} />
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(() => {
        expect(getByTestId('UNX-pageSize__dropdown')).toBeInTheDocument();
        expect(getByText('5')).toBeInTheDocument();
        expect(getByText('10')).toBeInTheDocument();
        expect(getByText('15')).toBeInTheDocument();
        expect(
            container.getElementsByClassName('UNX-productCard__container')
        ).toHaveLength(10);
    });
});

test('Page size options test', async () => {
    const sizeOptions = [
        { id: 50, value: '50' },
        { id: 100, value: '100' }
    ];
    const { getByText, getByTestId, container } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <PageSize sizeOptions={sizeOptions} />
                <SearchBox defaultSearch="shoes" />
                <Products attributesMap={attributesMap} />
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(() => {
        expect(getByTestId('UNX-pageSize__dropdown')).toBeInTheDocument();
        expect(getByText('50')).toBeInTheDocument();
        expect(getByText('100')).toBeInTheDocument();
    });
});

test('Page size component test', async () => {
    const PageSizeItemComponent = ({ itemData, onClick }) => {
        const { value, isSelected } = itemData;
        const handleClick = () => {
            onClick(itemData);
        };
        return (
            <button
                className={`UNX-pageSize__item ${
                    isSelected ? '-selected' : ''
                }`}
                data-testid="custom-page-size-btn"
                onClick={handleClick}
            >
                {value}
            </button>
        );
    };

    const { getAllByTestId } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <PageSize
                    displayType={'LIST'}
                    pageSizeItemComponent={<PageSizeItemComponent />}
                />
                <SearchBox defaultSearch="shoes" />
                <Products attributesMap={attributesMap} />
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(() => {
        expect(getAllByTestId('custom-page-size-btn'));
    });
});

test('Page size label test', async () => {
    const { getByText } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <PageSize label={'Products per page:'} />
                <SearchBox defaultSearch="shoes" />
                <Products attributesMap={attributesMap} />
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(() => {
        expect(getByText('Products per page:'));
    });
});

test('Page size size option test', async () => {
    const sizeOptions = [
        { id: 5, value: '5' },
        { id: 10, value: '10' },
        { id: 15, value: '15' }
    ];
    const { getByText, getByTestId, container } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <PageSize sizeOptions={sizeOptions} size={5} />
                <div className="hidden">
                    <SearchBox defaultSearch="shoes" />
                </div>
                <Products attributesMap={attributesMap} />
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(() => {
        expect(getByTestId('UNX-pageSize__dropdown')).toBeInTheDocument();
        expect(getByText('5')).toBeInTheDocument();
        expect(
            container.getElementsByClassName('UNX-productCard__container')
        ).toHaveLength(10);
    });
});
