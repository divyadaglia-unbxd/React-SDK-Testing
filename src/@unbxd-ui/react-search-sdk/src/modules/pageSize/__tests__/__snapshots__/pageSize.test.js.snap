// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Snapshot for page size 1`] = `
Array [
  <div
    className="UNX-pageSize__container"
  >
    <select
      className="UNX-pageSize__dropdown"
      data-testid="UNX-pageSize__dropdown"
      id="UNX-pageSize__dropdown"
      name="pageSize"
      onChange={[Function]}
      value={10}
    >
      <option
        value={5}
      >
        5
      </option>
      <option
        value={10}
      >
        10
      </option>
      <option
        value={15}
      >
        15
      </option>
    </select>
  </div>,
  <div
    className="UNX-searchbox__container"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="UNX-searchbox__input"
        onChange={[Function]}
        placeholder=""
        value="boots"
      />
      <button
        aria-label="search"
        className="UNX-searchbox__button"
        tabIndex={0}
        type="submit"
      >
        Search
      </button>
    </form>
  </div>,
  <div>
    <div
      className="UNX-products__container"
    >
      <div
        className="UNX-products__list -grid grid-cols-4"
      >
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/vasque-breeze-all-terrain-gtx-mens-hiking-boots-br/V7040W"
            onClick={[Function]}
          >
            <img
              alt="Vasque Breeze All-Terrain GTX Mens Hiking Boots - Brown Olive/Bossa Nova"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/V7040W.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Vasque Breeze All-Terrain GTX Mens Hiking Boots - Brown Olive/Bossa Nova
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                369.99
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/vasque-breeze-all-terrain-gtx-womens-hiking-boots/V7023W"
            onClick={[Function]}
          >
            <img
              alt="Vasque Breeze All-Terrain GTX Womens Hiking Boots - Gargyle"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/V7023W.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Vasque Breeze All-Terrain GTX Womens Hiking Boots - Gargyle
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                369.99
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-phantom-6000-mountaineering-boots-black-ora/SCA40017-BlackOrange"
            onClick={[Function]}
          >
            <img
              alt="Scarpa Phantom 6000 Unisex Waterproof Mountaineering Boots"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA40017-BlackOrange.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Phantom 6000 Unisex Waterproof Mountaineering Boots
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                1099.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-mont-blanc-pro-gtx-goretex-mountaineering-b/SCA40013"
            onClick={[Function]}
          >
            <img
              alt="Scarpa Mont Blanc Pro GTX Goretex Unisex Mountaineering Boots"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA40013.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Mont Blanc Pro GTX Goretex Unisex Mountaineering Boots
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                749.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-terra-gtx-unisex-shoes-brown/SCA00108"
            onClick={[Function]}
          >
            <img
              alt="Scarpa Terra Goretex Unisex Waterproof Hiking Boots - Brown"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA00108.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Terra Goretex Unisex Waterproof Hiking Boots - Brown
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                349.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-delta-leather-fly-mens-hiking-boots-mens-hi/SCA00090"
            onClick={[Function]}
          >
            <img
              alt="Scarpa Delta Leather Fly Mens Waterproof Hiking Boots - T.Di Moro"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA00090.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Delta Leather Fly Mens Waterproof Hiking Boots - T.Di Moro
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                399.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-delta-womens-leather-waterproof-hiking-boot/SCA00065"
            onClick={[Function]}
          >
            <img
              alt="Scarpa Delta Womens Goretex Waterproof Leather Hiking Boots"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA00065.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Delta Womens Goretex Waterproof Leather Hiking Boots
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                499.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/salomon-quest-prime-gtx-womens-hiking-boots-ebony/411299"
            onClick={[Function]}
          >
            <img
              alt="Salomon Quest Prime GTX Womens Hiking Boots - Ebony/Black/Icy Morn"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/411299.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Salomon Quest Prime GTX Womens Hiking Boots - Ebony/Black/Icy Morn
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                299.99
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/salomon-outblast-ts-cswp-mens-hiking-boots-black-b/409223"
            onClick={[Function]}
          >
            <img
              alt="Salomon OUTblast TS CSWP Mens Hiking Boots - Black/Black/Black"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/409223.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Salomon OUTblast TS CSWP Mens Hiking Boots - Black/Black/Black
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                259.99
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/salomon-quest-prime-gtx-mens-hiking-shoes-phantom/404637"
            onClick={[Function]}
          >
            <img
              alt="Salomon Quest Prime Goretex Mens Waterproof Hiking Boots - Phantom/Black/Quiet Shade"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/404637.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Salomon Quest Prime Goretex Mens Waterproof Hiking Boots - Phantom/Black/Quiet Shade
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                299.99
                 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
